File: 3.0/golang/11.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type stack struct {
	arr []int
	ln  int
}

func (stack *stack) append(x ...int) {
	for _, v := range x {
		stack.arr = append(stack.arr, v)
		stack.ln += 1
	}
}

func (stack *stack) pop() int {
	num := stack.arr[stack.ln-1]
	stack.arr = stack.arr[:stack.ln-1]
	stack.ln -= 1

	return num
}

func (stack *stack) clear() {
	stack.arr = stack.arr[:0]
	stack.ln = 0
}

func main() {
	var slice []int
	stack := stack{arr: slice, ln: 0}

	in := bufio.NewReader(os.Stdin)

Loop:
	for true {
		cmdb, _, _ := in.ReadLine()
		cmd := strings.Split(string(cmdb), " ")

		switch {
		case cmd[0] == "push":
			num, _ := strconv.Atoi(cmd[1])
			stack.append(num)
			fmt.Println("ok")

		case cmd[0] == "pop":
			if stack.ln == 0 {
				fmt.Println("error")
				continue
			}
			fmt.Println(stack.pop())

		case cmd[0] == "back":
			if stack.ln == 0 {
				fmt.Println("error")
				continue
			}
			fmt.Println(stack.arr[stack.ln-1])

		case cmd[0] == "size":
			fmt.Println(stack.ln)

		case cmd[0] == "clear":
			stack.clear()
			fmt.Println("ok")

		case cmd[0] == "exit":
			fmt.Println("bye")
			break Loop
		}
	}
}

File: 3.0/python/11.py
stack = []
ln = 0

while True:
    cmd = input()

    if "push" in cmd:
        cmd, n = cmd.split()
        stack.append(n)
        ln += 1
        print("ok")

    if cmd == "pop":
        if ln == 0:
            print("error")
            continue
        print(stack.pop())
        ln -= 1

    if cmd == "back":
        if ln == 0:
            print("error")
            continue
        print(stack[-1])
    
    if cmd == "size":
        print(ln)
    
    if cmd == "clear":
        stack.clear()
        ln = 0
        print("ok")

    if cmd == "exit":
        print("bye")
        break